tests/test_pipeline.py

```diff
@@ -150,12 +155,33 @@ def testParameters(self):
 
     def testSerialization(self):
         pipeline = Pipeline("test")
-        pipeline.addTask(AddTask, "add")
         pipeline.addTask(MultTask, "mult")
+        pipeline.addTask(AddTask, "add")
+
+        # verify the keys are in insertion order
+        self.assertEqual(list(pipeline._pipelineIR.tasks.keys()), ['mult', 'add'])
```

Comment:
I assume you're using internal APIs in these tests because the public way of getting the keys is going to sort them automatically?  If so, it might be better to remove these "before" checks and just verify that the public order is topological, since that should be the only thing the interface actually guarantees.

Reply:
Yeah, that is exactly why I was doing it, I wanted to definitively check that they were unsorted before, to guard against some future change where someone makes a change and suddenly the test is passing because of ordering, but where the sorting is broken.